browser/main/modals/PreferencesModal/UiTab.js

```diff
@@ -91,8 +90,19 @@ class UiTab extends React.Component {
     if (newCodemirrorTheme !== codemirrorTheme) {
       checkHighLight.setAttribute('href', `../node_modules/codemirror/theme/${newCodemirrorTheme.split(' ')[0]}.css`)
     }
-
-    this.setState({ config: newConfig, codemirrorTheme: newCodemirrorTheme })
+    this.setState({ config: newConfig, codemirrorTheme: newCodemirrorTheme }, () => {
+      const {ui, editor, preview} = this.props.config
+      this.currentConfig = {ui, editor, preview}
+      if (JSON.stringify(this.currentConfig) === JSON.stringify(this.state.config)) {
```

Comment:
Couldn't you write this logic without `JSON.stringify`? It will be overhead.
If I try to write this logic, I will use `_.isEqual`.
https://lodash.com/docs/4.17.4#isEqual

Reply:
Yes, I can use ```isEqual``` from lodash, it is a more robust solution than ```JSON.stringify```.